webui/webui/components/navbar.py

```diff
@@ -1,68 +1,124 @@
 import reflex as rx
-
-from webui import styles
 from webui.state import State
 
+def sidebar_chat(chat: str) -> rx.Component:
+    """A sidebar chat item.
 
-def navbar():
-    return rx.chakra.box(
-        rx.chakra.hstack(
-            rx.chakra.hstack(
-                rx.chakra.icon(
-                    tag="hamburger",
-                    mr=4,
-                    on_click=State.toggle_drawer,
-                    cursor="pointer",
+    Args:
+        chat: The chat item.
+    """
+    return rx.hstack(
+        rx.button(
+            chat, on_click=lambda: State.set_chat(chat), width="80%", variant="surface"
+        ),
+        rx.button(
+            rx.icon(
+                tag="trash",
+                on_click=State.delete_chat,
+                stroke_width=1,
+            ),
+            width="20%",
+            variant="surface",
+            color_scheme="red",
+        ),
+        width="100%",
+    )
+
+
+def sidebar(trigger) -> rx.Component:
+    """The sidebar component."""
+    return rx.drawer.root(
+        rx.drawer.trigger(trigger),
+        rx.drawer.overlay(),
+        rx.drawer.portal(
+            rx.drawer.content(
+                rx.vstack(
+                    rx.heading("Chats", color=rx.color("mauve", 11)),
+                    rx.divider(),
+                    rx.foreach(State.chat_titles, lambda chat: sidebar_chat(chat)),
+                    align_items="stretch",
+                    width="100%",
                 ),
-                rx.chakra.link(
-                    rx.chakra.box(
-                        rx.chakra.image(src="favicon.ico", width=30, height="auto"),
-                        p="1",
-                        border_radius="6",
-                        bg="#F0F0F0",
-                        mr="2",
-                    ),
-                    href="/",
+                top="auto",
+                right="auto",
+                height="100%",
+                width="20em",
+                padding="2em",
+                background_color=rx.color("mauve", 2),
+                outline="none",
+            )
+        ),
+        direction="left",
+    )
+
+
+def modal(trigger) -> rx.Component:
+    """A modal to create a new chat."""
+    return rx.dialog.root(
+        rx.dialog.trigger(trigger),
+        rx.dialog.content(
+            rx.hstack(
+                rx.input(
```

Comment:
This should go in an `rx.form` once the overlay issues are fixed

Reply:
There is no need to put it in a form here as its just one input

Reply:
Mostly so you can hit enter to submit instead of clicking the button